### PR TITLE
release: prepare for release v0.0.7

### PR DESCRIPTION
**Pull request by flywukong**
### Description

1.  update dependency and readme
2.  support parallel hash computing before creating object in the client .

### Rationale
support new API to compute Integrity Hashe

### Example

NA

### Changes

Notable changes:
* https://github.com/bnb-chain/greenfield-common/pull/29  feat: support parallel hash computing

### CODE REVIEW

File: go/hash/hash.go [hunk 1]
```diff
@@ -146,9 +153,21 @@ func (i *IntegrityHasher) computeBufferHash() error {
...
 
-// ComputeIntegrityHash split the reader into segment, ec encode the data, compute the hash roots of pieces
-// return the hash result array list and data segmentSize
-func ComputeIntegrityHash(reader io.Reader, segmentSize int64, dataShards, parityShards int) ([][]byte, int64,
+// ComputeIntegrityHash  return the integrity hash of file and data size
+// If isSerial is true, compute the integrity hash using the serial version
+// If isSerial is false or not provided, compute the integrity hash using the parallel version
+func ComputeIntegrityHash(reader io.Reader, segmentSize int64, dataShards, parityShards int, isSerial bool) ([][]byte, int64,
+	storageTypes.RedundancyType, error,
+) {
```
Comment (sysvm) [inline go/hash/hash.go:161]: move 161 rows to 160 rows

File: go/hash/hash.go [hunk 2]
```diff
@@ -146,9 +153,21 @@ func (i *IntegrityHasher) computeBufferHash() error {
...
+		return ComputeIntegrityHashSerial(reader, segmentSize, dataShards, parityShards)
+	}
+	return ComputeIntegrityHashParallel(reader, segmentSize, dataShards, parityShards)
+}
+
+// ComputeIntegrityHashSerial split the reader into segment, ec encode the data, compute the hash roots of pieces in a serial way
+// return the hash result array list and data size
+func ComputeIntegrityHashSerial(reader io.Reader, segmentSize int64, dataShards, parityShards int) ([][]byte, int64,
 	storageTypes.RedundancyType, error,
 ) {
```
Comment (sysvm) [inline go/hash/hash.go:172]: ditto

File: go/hash/hash.go [hunk 3]
```diff
@@ -221,11 +249,150 @@ func ComputerHashFromFile(filePath string, segmentSize int64, dataShards, parity
...
+	}
+
+	return pieceChecksumList, nil
+}
+
+// hashWorker receive the segment info and compute the corresponding segment hash and piece hashes.
+// The result will be stored in the sync map to compute integrity hash in order.
+func hashWorker(jobs <-chan SegmentInfo, errChan chan<- error, dataShards, parityShards int, wg *sync.WaitGroup,
+	segmentHashMap *sync.Map, pieceHashMap *sync.Map,
+) {
```
Comment (sysvm) [inline go/hash/hash.go:283]: ditto

File: go/hash/hash.go [hunk 4]
```diff
@@ -221,11 +249,150 @@ func ComputerHashFromFile(filePath string, segmentSize int64, dataShards, parity
...
+	defer wg.Done()
+
+	for segInfo := range jobs {
+		checksum := GenerateChecksum(segInfo.Data)
+		segmentHashMap.Store(segInfo.SegmentID, checksum)
+
+		pieceCheckSumList, err := computePieceHashes(segInfo.Data, dataShards, parityShards)
```
Comment (sysvm) [inline go/hash/hash.go:290]: fix `pieceCheckSumList` as `pieceChecksumList `

Comment (flywukong) [reply go/hash/hash.go]: fixed

File: go/hash/hash.go (hunk 1 above)
Comment (flywukong) [reply go/hash/hash.go:161]: This format conforms to the lint specification

**Review by sysvm — APPROVED** [review]
LGTM

**Pull request merged**